src/Uart/Uart.c

```diff
@@ -334,79 +317,81 @@ uint32_t
 Uart_getRxFifoCount(Uart* const uart)
 {
     rtems_interrupt_vector_disable(interruptNumber(uart->id));
-    bool result = ByteFifo_getCount(uart->rxFifo);
+    uint32_t result = ByteFifo_getCount(uart->rxFifo);
     rtems_interrupt_vector_enable(interruptNumber(uart->id));
     return result;
 }
 
 bool
 Uart_handleError(Uart* const uart)
 {
-    if(Uart_getLinkErrors(uart->reg->status, &uart->errorFlags) == true) {
+    bool result = false;
+
+    if (Uart_getLinkErrors(uart->reg->status, &uart->errorFlags) == true) {
         uart->errorHandler.callback(uart->errorHandler.arg);
-        return true;
+        result = true;
     }
-    return false;
+
+    return result;
 }
 
 bool
 Uart_handleRx(Uart* const uart)
 {
-    if((uart->reg->control & UART_CONTROL_RE) != 0
-       && (uart->reg->status & UART_STATUS_DR) != 0) {
-        if(uart->rxFifo != NULL) {
+    bool result = false;
+
+    if (Uart_getFlag(uart->reg->control, UART_RE) && Uart_getFlag(uart->reg->status, UART_DR)) {
+        if (uart->rxFifo != NULL) {
             uint8_t buf = uart->reg->data;
-            if(ByteFifo_isFull(uart->rxFifo) == true) {
+            if (ByteFifo_isFull(uart->rxFifo)) {
                 uart->errorFlags.hasRxFifoFullErrorOccurred = true;
-                return true;
-            }
-            if(buf == uart->rxHandler.targetCharacter) {
-                uart->rxHandler.characterCallback(uart->rxHandler.characterArg);
-            }
-            uart->interruptData.sentBytes++;
-            if(uart->interruptData.sentBytes == uart->rxHandler.targetLength) {
-                uart->interruptData.sentBytes = 0;
-                uart->rxHandler.lengthCallback(uart->rxHandler.lengthArg);
+                result = true;
+            } else {
+                if (buf == uart->rxHandler.targetCharacter) {
+                    uart->rxHandler.characterCallback(uart->rxHandler.characterArg);
+                }
+                uart->interruptData.sentBytes++;
+                if (uart->interruptData.sentBytes == uart->rxHandler.targetLength) {
+                    uart->interruptData.sentBytes = 0;
+                    uart->rxHandler.lengthCallback(uart->rxHandler.lengthArg);
+                }
+                ByteFifo_push(uart->rxFifo, buf);
             }
-            ByteFifo_push(uart->rxFifo, buf);
         }
-        return true;
+        result = true;
     }
-    return false;
+
+    return result;
```

Comment:
Out of curiosity, why are we discarding early returns?

Reply:
OK... I will close this comment, but please don't do this in the future. Early returns are a good practice that we should try to follow.